### ISSUE TITLE
Fit text toggle disappears when toggled to 'off'

### ISSUE THREAD

**Issue by joedolson** [WordPress/gutenberg#76691]
### Description

When the fit text typography option is enabled, the toggle appears in the settings panel, as expected. When you turn it on, it behaves as expected. However, if you turn it off, the option disappears from the panel, causing a loss of focus.

This makes it difficult to experiment with the option - as if you turn it off, you need to navigate back to the typography panel to re-add the control, then navigate to the control to enable it.

It behaves differently from any other control in the Typography panel - none of the other controls are binary settings, and there is no "off" setting, just variants. I suspect this is an incidental result of handling of a false value in the setting that doesn't impact other controls.

### Step-by-step reproduction instructions

1. Add a paragraph or heading.
2. Navigate to Typography settings.
3. Add the 'Fit text' option.
4. Turn on the fit text option.
5. Turn off the fit text option.

The option is removed from the panel.

### Screenshots, screen recording, code snippet

https://github.com/user-attachments/assets/38f9173e-f3d0-498f-8372-f51000dfa116

### Environment info

- 7.0-beta5-62067
- Tested with and without Gutenberg 22.7.1

### Please confirm that you have searched existing issues in the repo.

- [x] Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

- [x] Yes

### Please confirm which theme type you used for testing.

- [x] Block
- [ ] Classic
- [ ] Hybrid (e.g. classic with theme.json)
- [ ] Not sure

**Comment by t-hamano** [comment]
Thanks for the report. This problem should be fixed in https://github.com/WordPress/gutenberg/pull/75092
